llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -229,6 +236,9 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
 
   getActionDefinitionsBuilder(G_ICMP)
       .legalFor({{sXLen, sXLen}, {sXLen, p0}})
+      .legalIf(all(typeIsLegalBoolVec(0, BoolVecTys, ST),
+                   typeIsLegalIntOrFPVec(1, IntOrFPVecTys, ST)))
+      .widenScalarOrEltToNextPow2OrMinSize(1, 8)
```

Comment:
Does this make the `.widenScalarToNextPow2(1)` on the next line redundant?

Reply:
With `widenScalarToNextPow2` if the size was s1, then it would not be widened. There is nothing in the MachineVerifier that says it couldn't be an i1 scalar.

Reply:
Doesn't MinSize there mean widen to next power of 2 that is >= 8?

Reply:
The description says

> Widen the scalar to the next power of two that is at least MinSize. No effect if the type is not a scalar or is a power of two.

I took

> no effect if the type is not a ... power of two

to mean that exactly that.

I think you took it as

> no effect if the type is not a .. power of two or at least MinSize

I will check how it actually works.

Reply:
Please fix the comment(as a separate patch) once you figure it out. The version without `OrMinSize` suffix has the same description.

Reply:
Removed the redundancy here and pushed an NFC commit fixing the docstring in 1d5e5f4d3c68e63ced47ee9b17d62fb995aa1e62